### DISCUSSION TITLE
FortiClient VPN is not up to date in Winget repository

### DISCUSSION THREAD

**Discussion by Swe-it** [microsoft/winget-cli#2978] [General]
FortiClient VPN is very out of date, winget offers 6.2 where Fortinet has a 7.0.7 version.

**Comment by ItzLevvie** [comment]


**Reply by Swe-it** [reply]
That thread is 1 year old and nothing has happend, it's the 2019 version you get when you install it through Winget.

**Reply by Swe-it** [reply]
I have manage to install 7.0.5 silent, havn't tried 7.0.7 yet

**Reply by denelon** [reply]
I'd suggest submitting the manifests as PRs so we can add "Blocking-Label" and "Interactive-Only-Installer" so we can see the demand for the work to isolate interactive only installers from automated scripts, and still allow them in the community repository.

There are only a few PRs like this today: https://github.com/microsoft/winget-pkgs/pulls?q=is%3Aopen+is%3Apr+label%3AInteractive-Only-Installer

We're tracking the work to support this with:
* https://github.com/microsoft/winget-cli/issues/823

**Reply by Swe-it** [reply]
This line works on 7.0.5.0238

`msiexec /i "FortiClientVPN.msi" /passive /quiet INSTALLLEVEL=3 /NORESTART`

**Reply by jantari** [reply]
The MSI version of FortiClient is easy to install silently, but I believe the problem here is that the MSI download is only available if you have a Fortinet Support Subscription - it's not public, so winget cannot download and use it in the public repository.

**Reply by Swe-it** [reply]
Someone will have to extract the MSI from the EXE and upload it to the repository. 
I dont think Fortinet has for a long time been having an MSI available, only exe.

**Reply by Masamune3210** [reply]
Winget does not store installers, only links to them for the client to download. If its not publically accessible you are likely out of luck unfortunately

**Reply by jantari** [reply]
Yes, it's not possible in the public repository because it requires the download links there to be accessible. You have to host your own winget package source if you want to distribute your MSI forticlient installer.
